Client/Frontend/Menu/MenuView.swift

```diff
@@ -0,0 +1,387 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+class MenuView: UIView {
+
+    var toolbar: RoundedToolbar
+
+    var openMenuImage: UIImageView
+    var menuFooterView: UIView
+
+    var toolbarDelegate: MenuToolbarItemDelegate?
+    var toolbarDataSource: MenuToolbarDataSource?
+    var menuItemDelegate: MenuItemDelegate?
+    var menuItemDataSource: MenuItemDataSource?
+
+    private let menuItemPageController = MenuPagingViewController()
+    private var presentationStyle: MenuViewPresentationStyle
+
+    private var menuColor: UIColor = UIColor.clearColor() {
+        didSet {
+            menuItemPageController.backgroundColor = menuColor
+            menuFooterView.backgroundColor = menuColor
+        }
+    }
+
+    override var backgroundColor: UIColor! {
+        get { return menuColor }
+
+        set { menuColor = newValue }
+    }
+    
+    override var tintColor: UIColor! {
+        didSet {
+            menuItemPageController.tintColor = tintColor
+        }
+    }
+
+    var toolbarHeight: Float = 40.0 {
+        didSet {
+            self.setNeedsLayout()
+        }
+    }
+
+    var menuRowHeight: Float = 65.0 {
+        didSet {
+            self.setNeedsLayout()
+        }
+    }
+
+    var itemPadding: CGFloat = 10.0 {
+        didSet {
+            self.setNeedsLayout()
+        }
+    }
+
+    var currentPageIndex: Int = 0 {
+        didSet {
+            needsReload = true
+            self.setNeedsLayout()
+        }
+    }
+
+    var menuFooterHeight: CGFloat = 40 {
+        didSet {
+            self.setNeedsLayout()
+        }
+    }
+
+    var nextPageIndex: Int?
+
+    private var needsReload: Bool = true
+
+    private var cachedItems = [NSIndexPath: MenuItemView]()
+    private var reusableItems = [MenuItemView]()
+    private var toolbarItems = [UIBarButtonItem]()
+
+    init(presentationStyle: MenuViewPresentationStyle) {
+        self.presentationStyle = presentationStyle
+        toolbar = RoundedToolbar()
+        toolbar.setContentHuggingPriority(UILayoutPriorityRequired, forAxis: UILayoutConstraintAxis.Vertical)
+        menuFooterView = UIView()
+        openMenuImage = UIImageView()
+        openMenuImage.contentMode = UIViewContentMode.ScaleAspectFit
+
+        super.init(frame: CGRectZero)
+
+
+        let pageControllerView = menuItemPageController.view
```

Comment:
If you're adding the `menuItemPageController`'s view as a subview, I think you also should be following the UIViewControllerContainer steps to adding the view controller as a child as well [1]

[1] https://developer.apple.com/library/ios/featuredarticles/ViewControllerPGforiPhoneOS/ImplementingaContainerViewController.html